### PR TITLE
Create Toggle component

### PR DESCRIPTION
**Pull request by guilean**
**Checklist**
- [X] New feature (non-breaking change which adds functionality)

**Please check if the PR fulfills these requirements**
- [X] Tests for the changes have been added (for bug fixes / features)

Screenshot 2020-12-16 at 15 39 49
Screenshot 2020-12-16 at 15 39 53

### CODE REVIEW

File: packages/components/src/components/primitives/Toggle/Toggle.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  component: Toggle,
+};
+
+export const Playground = () => (
+  <Toggle
+    onStateChanged={action('on state changed')}
+    enabled={boolean('Enabled', false)}
+    disabled={boolean('Disabled', false)}
+  />
+);
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Toggle/Toggle.stories.tsx:17]: we should have both states in a component to allow chromatic to catch regressions

File: packages/components/src/components/primitives/Toggle/Toggle.styled.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+} from 'components/Components.styled.types';
+import { getTheme } from 'utils/getTheme';
+
+type Props = StyledBaseProps & StyledCssProps;
+
+export const StyledLabel = styled.label`
+  position: relative;
+  display: inline-block;
+  width: 42px;
+  height: 24px;
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Toggle/Toggle.styled.ts:14]: rem? (applies to the whole file)

File: packages/components/src/components/primitives/Toggle/Toggle.types.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { BaseProps } from 'components/Components.types';
+
+export type Props = BaseProps & {
+  enabled?: boolean;
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Toggle/Toggle.types.ts:4]: on instead?, having enabled and disabled as properties ...

File: packages/components/src/styles/colors.ts [hunk 4]
```diff
@@ -94,7 +94,7 @@ export const colors = {
   fillTransparent200: '#00000026',
   fillTransparent100: '#00000000',
   fillOverlay: '#00000066',
-  fillStatePrimaryEnabled: '#dd0303ff',
+  fillStatePrimaryEnabled: '#414141',
```
Comment (tkanzakic) [inline packages/components/src/styles/colors.ts:97]: double checking: is this correct?

Comment (guilean) [reply packages/components/src/styles/colors.ts:97]: Yes, I double checked DSM and state primary enabled is #414141 not #dd0303ff

File: packages/components/src/components/primitives/Toggle/Toggle.styled.ts (hunk 2 above)
Comment (guilean) [reply packages/components/src/components/primitives/Toggle/Toggle.styled.ts:14]: Changed to `rem` units

File: packages/components/src/components/primitives/Toggle/Toggle.stories.tsx (hunk 1 above)
Comment (guilean) [reply packages/components/src/components/primitives/Toggle/Toggle.stories.tsx]: Added default stories

File: packages/components/src/components/primitives/Toggle/Toggle.types.ts (hunk 3 above)
Comment (guilean) [reply packages/components/src/components/primitives/Toggle/Toggle.types.ts]: Renamed

File: packages/components/src/components/primitives/Toggle/Toggle.stories.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+import { Toggle } from '.';
+import { boolean } from '@storybook/addon-knobs';
+import { action } from '@storybook/addon-actions';
+
+export default {
+  title: 'Primitives/Toggle',
+  component: Toggle,
+};
+
+export const Enabled = () => <Toggle on />;
```
Comment (tkanzakic) [inline packages/components/src/components/primitives/Toggle/Toggle.stories.tsx:11]: rename to On. Also missing Off case

**Review by tkanzakic — CHANGES_REQUESTED** [review]

File: packages/components/src/components/primitives/Toggle/Toggle.stories.tsx (hunk 5 above)
Comment (guilean) [reply packages/components/src/components/primitives/Toggle/Toggle.stories.tsx]: updated

**Review by tkanzakic — APPROVED** [review]

**Pull request merged**